kernel/kernel-api/src/main/java/io/delta/kernel/expressions/StartsWith.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.expressions;
+
+import io.delta.kernel.annotation.Evolving;
+
+/**
+ * {@code STARTS WITH} expression
+ * <p>
+ * Definition:
+ * <p><ul>
+ *     <li>{@code expr1} STARTS WITH {@code expr2} on two inputs.</li>
+ *     <li>Requires both left and right input expressions of type {@link Predicate}.</li>
```

Comment:
left and right need to be strings, right?

Reply:
yes. updated.